cmd/loadContract.go

```
@@ -0,0 +1,166 @@
+			if err == nil && (stdinInfo.Mode()&os.ModeCharDevice) == 0 {
+				useStdin = true
+			}
+		}
+
+		// Process based on input source
+		if cfg.LoadContractConfig.Batch && useFile {
+			// Load contracts from file
+			err := contractStore.InitializeExternalContracts(filename)
+			if err != nil {
+				return fmt.Errorf("failed to initialize external contracts from file: %w", err)
+			}
+			return nil
+		} else if cfg.LoadContractConfig.Batch && useStdin {
+			// Read directly from stdin
+			err := contractStore.InitializeExternalContractsFromReader(os.Stdin)
+			if err != nil {
+				return fmt.Errorf("failed to initialize external contracts from stdin: %w", err)
+			}
+			return nil
+		} else if cfg.LoadContractConfig.Batch {
+			// If batch mode is enabled but no input source is provided
+			return fmt.Errorf("batch mode requires a file path or stdin input")
+		} else {
+			// If no file or stdin is provided, use the individual contract parameters
+			blockNumber := cfg.LoadContractConfig.BlockNumber
+			address := cfg.LoadContractConfig.Address
+			abi := cfg.LoadContractConfig.Abi
+			bytecodeHash := cfg.LoadContractConfig.BytecodeHash
+			associateToProxy := cfg.LoadContractConfig.AssociateToProxy
+
+			// Validate required parameters
+			if address == "" {
+				return fmt.Errorf("contract address is required in --contract.address")
+			} else if abi == "" {
+				return fmt.Errorf("contract ABI is required with the contract address in --contract.abi")
+			}
+			// Fetch bytecode hash if not provided
+			if bytecodeHash == "" {
+				bytecodeHash, err = af.FetchContractBytecodeHash(ctx, address)
+				if err != nil {
+					return fmt.Errorf("failed to fetch contract bytecode hash: %w", err)
+				}
+			}
+			// Associate to proxy if specified
+			if associateToProxy != "" {
+				_, err := contractStore.GetContractForAddress(associateToProxy)
+				if err != nil {
+					return fmt.Errorf("proxy contract %s not found: %w", associateToProxy, err)
```

If you look carefully at `GetContractForAddress`, we actually return `nil, nil` for a `ErrRecordNotFound` response, so the error in this case is something else. You'll need to add a check to see that the returned contract is `nil` to satisfy the not found vs. found check

---

If it is nil, I would return an error telling the caller that the proxy they want to associate to doesnt exist

---

Only checking if the contract is nil since the more error message will also show from `GetContractForAddress`